include/odp/api/spec/ipsec.h

```diff
@@ -1715,6 +1782,35 @@ int odp_ipsec_result(odp_ipsec_packet_result_t *result, odp_packet_t packet);
  */
 int odp_ipsec_status(odp_ipsec_status_t *status, odp_event_t event);
 
+/**
+ * IPSEC test API for modifying internal state of an SA.
+ *
+ * This function is not meant to be used by normal applications but by special
+ * test applications that test or debug the operation of the underlying ODP
+ * implementation. Calling this function may degrade the performance of the
+ * calling thread, other threads or the IPSEC implementation in general.
+ *
+ * Calling this function for an SA at the same time when the SA is used for
+ * processing traffic or when the SA is being modified through other parts
+ * of IPSEC API may result in undefined behaviour.
+ *
+ * SA state update through this function may not be supported by all ODP
+ * implementations, ODP instances or SA instances or at every moment. This
+ * function may return failure for unspecified reasons even when the capability
+ * call indicated support for updating a particular parameter and previous
+ * similar calls succeeded.
+ *
+ * @param          sa            IPSEC SA to be updated
+ * @param          op            Specifies operation to be performed
+ * @param          param         IPSEC SA param to be used for the operation
+ *
+ * @return 0      On success
+ * @retval <0     On failure
+ */
+int odp_ipsec_test_sa_update(odp_ipsec_sa_t sa,
+			     odp_ipsec_test_sa_operation_t op,
+			     odp_ipsec_test_sa_param_t param);
```

Comment:
I wonder if the param struct should be passed by reference rather than by value?

Reply:
Agreed. I will update this in v7.